spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/RowWriterFactory.scala

```diff
@@ -13,7 +13,7 @@ trait RowWriterFactory[T] {
   /** Creates a new `RowWriter` instance.
     * @param table target table the user wants to write into
     * @param columnNames columns selected by the user; the user might wish to write only a subset of columns */
-  def rowWriter(table: TableDef, columnNames: Seq[String]): RowWriter[T]
+  def rowWriter(table: TableDef, columnNames: Seq[String], checkColumns: CheckSetting): RowWriter[T]
```

Comment:
Whenever I need to add something like this to optionally disable  something somewhere, I think whether the thing I want to disable was in the right place at all. Maybe rowWriters should not perform those checks? Maybe we should just move them somewhere else (TableWriter?) where they do not collide with the function you need this for. This might be reminiscent of ancient pre 1.0 times when there was no RowWriters at all and that logic was part of the TableWriter; then RowWrites have been extracted from TableWriter with those checks.